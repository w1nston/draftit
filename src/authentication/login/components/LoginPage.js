import React, { Component, Fragment } from 'react';
import { css } from 'emotion';

const loginFormStyle = css`
  display: flex;
  flex-direction: column;
`;

const inputStyle = css`
  background-color: #fff;
  border: 1px solid #bbb;
  border-radius: 4px;
  box-shadow: 0 2px 0 #ececec inset;
  box-sizing: border-box;
  color: #333;
  font-size: 0.875rem;
  height: 42px;
  margin-bottom: 8px;
  padding: 0 8px;

  @media(min-width: 991px) {
    width: 250px;
  }
`;

const loginButtonStyle = css`
  background-color: #14ad5b;
  color: #fbfbfd;
  font-size: 1.125rem;
  font-weight: 700;
  height: 42px;
  width: 100%;

  @media(min-width: 991px) {
    width: 250px;
  }
`;

class LoginPage extends Component {
  state = { email: '', password: '' };

  handleEmailChange = event => {
    this.setState({ email: event.target.value });
  };

  handlePasswordChange = event => {
    this.setState({ password: event.target.value });
  };

  clearFields = () => {
    this.setState({ email: '', password: '' });
  };

  handleLogin = event => {
    event.preventDefault();
    const { email, password } = this.state;

    if (email === '' || password === '') {
      return;
    }

    this.props.onSubmit && this.props.onSubmit({ email, password });
    
    this.clearFields();
  };

  render() {
    const { email, password } = this.state;

    return (
      <Fragment>
        <h2>Login</h2>
        <form onSubmit={this.handleLogin} className={loginFormStyle} method="POST">
          <label htmlFor="email">Email</label>
          <input
            className={inputStyle}
            type="email"
            value={email}
            onChange={this.handleEmailChange}
            name="email"
            placeholder="Email"
          />
          <label htmlFor="password">Password</label>
          <input
            className={inputStyle}
            type="password"
            value={password}
            onChange={this.handlePasswordChange}
            name="password"
            placeholder="Password"
          />
          <button className={loginButtonStyle} type="submit">Login</button>
        </form>
      </Fragment>
    );
  }
}

export default LoginPage;
