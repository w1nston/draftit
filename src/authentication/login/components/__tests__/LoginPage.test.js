import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from '../LoginPage';

describe('LoginPage', () => {
  it('renders', () => {
    const component = shallow(<LoginPage />);
    expect(component.type()).toBe('form');
  });
});
