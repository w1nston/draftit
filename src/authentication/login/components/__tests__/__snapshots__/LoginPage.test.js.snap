// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders 1`] = `
<React.Fragment>
  <h2>
    Login
  </h2>
  <form
    className="css-1iyoj2o"
    method="POST"
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="css-1dn81gn"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="css-1dn81gn"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="css-1yrgme5"
      type="submit"
    >
      Login
    </button>
  </form>
</React.Fragment>
`;
